From ad090272c831b5c62f0b310c776e76e3b1ffc88d Mon Sep 17 00:00:00 2001
From: Iain Lane <iain.lane@canonical.com>
Date: Fri, 15 Nov 2013 14:17:03 +0000
Subject: [PATCH] Prevent the NEON check in configure from passing under
 aarch64.

The test verifies that the NEON C intrinsics work, but the rest of the
codebase uses lots of direct ARMv7 NEON assembly. The same intrinsics
work in A64, but the assembly is slightly different.

Prevent the check from passing so that we don't use this where it won't
work.

Patch originally by William Grant <wgrant@ubuntu.com>

https://bugzilla.gnome.org/show_bug.cgi?id=712367
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index e5c223f..0495371 100644
--- a/configure.ac
+++ b/configure.ac
@@ -190,6 +190,7 @@ AC_LINK_IFELSE([
 AC_LANG_PROGRAM([[
   #include <arm_neon.h>
   int32x4_t testfunc(int16_t *a, int16_t *b) {
+      asm volatile ("vmull.s16 q0, d0, d0" : : : "q0");
       return vmull_s16(vld1_s16(a), vld1_s16(b));
   }
 ]])],
-- 
1.8.4.3